src/Swarm/Game/Step.hs

```diff
@@ -920,6 +927,60 @@ execConst c vs s k = do
         -- Return the value returned by the hypothetical command.
         return $ Out v s k
       _ -> badConst
+    Async -> case vs of
+      [VString name, prog@(VDelay t _)] -> do
+        -- Get the named robot and current game state
+        r <- robotNamed name >>= (`isJustOrFail` ["There is no robot named ", name])
+        g <- get @GameState
+        -- -----------------------------------------------------------------------------
+        -- TODO: move this code out
+        -- Slightly edited copy of Unit.hs version of runCESK:
+        let runCESK' :: CESK -> StateT Robot (StateT GameState IO) (Either Exn Value)
+            runCESK' (Up exn _ []) = return (Left exn)
+            runCESK' cesk = case finalValue cesk of
+              Just (v, _) -> return (Right v)
+              Nothing -> do
+                cesk1 <- stepCESK cesk
+                ticks += 1
+                runCESK' cesk1
+        -- -----------------------------------------------------------------------------
+        let evalAs :: IO (Either Exn Value)
+            evalAs =
+              Out prog s [FApp (VCApp Force []), FExec]
+                & runCESK'
+                & flip evalStateT (r & systemRobot .~ True)
+                & flip evalStateT (g & creativeMode .~ True)
+
+        va <- sendIO $ Async.async evalAs
+
+        return $ Out (VAsync t $ GAsync va) s k
+      _ -> badConst
+    Await -> case vs of
+      [va@(VAsync t (GAsync a))] -> do
+        time <- use ticks
+        mv <- sendIO $ Async.poll a
+        return $ case mv of
+          Nothing -> Waiting (time + 8) (Out va s (FApp (VCApp Await []) : k))
```

Comment:
I think an `FExec` is needed, so that the `Await` is re-executed once it is applied.
```suggestion
          Nothing -> Waiting (time + 8) (Out va s (FApp (VCApp Await []) : FExec : k))
```